views/find_missing.py

```diff
@@ -92,9 +114,136 @@ def render_search_button(self):
         btn_layout.addWidget(self.search_button)
         self.addLayout(btn_layout)
 
-    def search(self):
-        # Search through fetched cards here
-        deck_id = self.deck_selection_enabled and self.deck_chooser.selectedId()
+    def render_search_preview(self):
+        search_preview_layout = QHBoxLayout()
+        search_text = QLabel()
+        search_text.setText("Search:")
+        search_preview_layout.addWidget(search_text)
+        
+        self.search_preview = QTextEdit()
+        self.search_preview.setReadOnly(True)
+        self.search_preview.setFixedHeight(40)
+        search_preview_layout.addWidget(self.search_preview)
+
+        # self.copy_preview_button = QPushButton("Copy")
+        # self.copy_preview_button.clicked.connect(self.copy_preview)
+        # search_preview_layout.addWidget(self.copy_preview_button)
+
+        self.addLayout(search_preview_layout)
+
+    def render_results_area(self):
+        self.results_area = QListWidget()
+        self.addWidget(self.results_area)
+
+
+
+########################
+#######  Non-UI  #######
+########################
+
+
+
+    # def copy_preview(self):
+    #     data = self.get_final_search("[Word]")
+    #     command = 'echo ' + data.strip() + '| clip'
+    #     os.system(command)
+    
+    def update_init_search(self):
+        # print("Update Init Search")
         deck_name = self.deck_selection_enabled and self.deck_chooser.deckName()
-        model_fields = self.model_field_selection_enabled and self.model_field_chooser.selected_items
+        model_fields = self.model_field_selection_enabled and self.model_field_selected_items
+
+        if self.model_field_selection_enabled:
+            noteTypesSelected = [mod['name'] for mod in model_fields] if model_fields else ""
+            fieldsSelected = {fields['name'] for mod in model_fields for fields in mod['fields']} if model_fields else ""
+            print(noteTypesSelected, fieldsSelected)
 
+        self.initQuery = ""
+        self.initQuery += self.search_formatter(True, "deck", deck_name) if deck_name else ""
+        self.initQuery += self.search_multiple_terms_formatter("note", noteTypesSelected) if self.model_field_selection_enabled else ""
+        self.initQuery += self.search_multiple_fields_formatter(fieldsSelected, self.REPLACEMENT_STRING) if self.model_field_selection_enabled else ""
+        
+        self.search_preview.setText(self.get_final_search("[Word]"))
+
+
+
+    def search(self):
+        # Search through fetched cards here
+        
+        self.results_area.clear()
+        self.results.clear()
+        
+        self.update_init_search()
+
+        words = {self.text_area_text.toPlainText().split()}
```

Comment:
This needs to be converted to `set(....split())`. Otherwise it thinks it's a dictionary literal setup incorrectly.

Caught exception:
  File "/home/dev/.local/share/Anki2/addons21/anki_LL/views/find_missing.py", line 178, in search
    words = {self.text_area_text.toPlainText().split()}
<class 'TypeError'>: unhashable type: 'list'

Reply:
`{ }` is "Set Comprehension" and returns a set. https://python-reference.readthedocs.io/en/latest/docs/comprehensions/set_comprehension.html

Reply:
Yes but you must still have a comprehension (variable for variable in iterable) inside, not just the iterable. It must either be set(list) or { i for i in list }.

Reply:
:thinking: I think this is another issue with your build, mate. `{}` with only one entry in it does not declare a hash, it is still a Set (with no Comprehension afterwards, yes). Googling around others who have your error `<class 'TypeError'>: unhashable type: 'list'` I see them declaring hashes in the form `{'name': 'value'}` with a colon separating. Their syntax is correct, but they are using a list as one of those values, which is not allowed.

My compiler still interprets this as a Set conversion and I get no errors or warnings, whereas yours seems to think it is a Hash, even though it lacks a colon and second variable :thinking: We should put some effort into working out where our build differences are coming from :sweat_smile: agreed?

Anyways I have changed it back to `words = set(self.text_area_text.toPlainText().split())` for now, to alleviate your issues, as I don't much mind how it looks :grin:

Reply:
Hmm, interesting. I wonder if something is different in our Python versions that Anki uses. I forget if it packages its own? Either way it does bring up the question of compatibility. I guess we'll just tackle them on a case-by-case basis.